modules/artifact/src/main/java/org/mule/module/artifact/classloader/DefaultResourceReleaser.java

```diff
@@ -187,24 +194,95 @@ private boolean hasDeclaredMethod(Class<?> clazz, String methodName, Class<?>...
    * Workaround for http://bugs.mysql.com/bug.php?id=65909
    */
   private void shutdownMySqlAbandonedConnectionCleanupThread() {
-
-    Class<?> classAbandonedConnectionCleanupThread;
     try {
-      classAbandonedConnectionCleanupThread =
-          this.getClass().getClassLoader().loadClass("com.mysql.jdbc.AbandonedConnectionCleanupThread");
+      Class<?> cleanupThreadsClass = findMySqlDriverClass();
+      if (cleanupThreadsClass == null) {
+        logger.warn("No AbandonedConnectionCleanupThread class found. Check if thread class is between "
+            + CONNECTION_CLEANUP_THREAD_KNOWN_CLASS_ADDRESES.toString());
+        throw new ClassNotFoundException(CONNECTION_CLEANUP_THREAD_KNOWN_CLASS_ADDRESES.get(0));
+      }
+      shutdownMySqlConnectionCleanupThreads(cleanupThreadsClass);
       try {
-        Method uncheckedShutdown = classAbandonedConnectionCleanupThread.getMethod("uncheckedShutdown");
-        uncheckedShutdown.invoke(null);
-      } catch (NoSuchMethodException e) {
-        Method checkedShutdown = classAbandonedConnectionCleanupThread.getMethod("shutdown");
-        checkedShutdown.invoke(null);
+        // The cleanup threads are fired from a single-thread ThreadPoolExecutor, which is created inside a
+        // lambda, which wraps the thread pool into a finalizable wrapper. This leads to retention in the
+        // artifact classloaders when several redeployments are performed.
+        cleanMySqlCleanupThreadsThreadFactory(cleanupThreadsClass);
+      } catch (NoSuchFieldException e) {
```

Comment:
no need for this double nested try/catch, move this block to the `cleanMySqlCleanupThreadsThreadFactory` method if want to log that this particular cleanup failed. 
Is more cohesive to handle the exception for logging in the method that has the logic that failed.

Reply:
Catch both exceptions, NoSuchFieldException and IllegalAccessException, since the factory cleanup can fail with any of those, and append the exception to the log. (`logger.warn("bla", e)`)

Reply:
Completely agree about the method cohesion